text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+  - Such resources may be provided via cloud systems, via emulation, or via
+    physical hardware.
+  - If the target requires the use of emulation to meet any of the tier
+    requirements, the approving teams for those requirements must have high
+    confidence in the accuracy of the emulation, such that discrepancies
+    between emulation and native operation that affect test results will
+    constitute a high-priority bug in either the emulation or the
+    implementation of the target.
+  - If it is not possible to run the target via emulation, these resources must
+    additionally be sufficient for the Rust infrastructure team to make them
+    available for access by Rust team members, for the purposes of development
+    and testing. (Note that the responsibility for doing target-specific
+    development to keep the target well maintained remains with the target
+    maintainers. This requirement ensures that it is possible for other
+    Rust developers to test the target, but does not obligate other Rust
+    developers to make target-specific fixes.)
+- New tier 1 targets must not have a hard requirement for signed, verified, or
+  otherwise "approved" binaries. Developers must be able to build, run, and
+  test binaries for the target on systems they control.  (Doing so may require
+  enabling some appropriate "developer mode" on such systems, but must not
+  require the payment of any additional fee or other consideration, or
+  agreement to any onerous legal agreements.)
+  - The Rust project may decide to supply appropriately signed binaries if
+    doing so provides a smoother experience for developers using the target,
+    and a tier 2 target with host tools already requires providing appropriate
+    mechanisms that enable our infrastructure to provide such signed binaries.
+    However, this additional tier 1 requirement ensures that Rust developers
+    can help develop and test Rust software for the target (including Rust
+    itself), and that development or testing for the target is not limited to
+    an exclusive group or entity.
+- All requirements for tier 2 apply.
+
+A tier 1 target may be demoted if it no longer meets these requirements but
+still meets the requirements for a lower tier. Any proposal for demotion of a
+tier 1 target requires a full RFC process, with approval by the compiler and
+release teams. Any such proposal will be communicated widely to the Rust
+community, both when initially proposed and before being dropped from a stable
+release. A tier 1 target is highly unlikely to be directly removed without
+first being demoted to tier 2 or tier 3.
+
+Raising the baseline expectations of a tier 1 target (such as the minimum CPU
+features or OS version required) requires the approval of the compiler and
+release teams, but does not necessarily require a full RFC.
```

Comment:
I think this is missing a section on requiring more than 2 maintainers. It is likely implicitly true due to the infrastructure provisioning points (though I'm unsure about those, but I think this is the more important aspect of this question).

In particular, I would largely expect that tier 1 targets -- due to the "substantial, widespread interest within the developer community" -- should have more than 2 developers willing to commit to being available for questions. I think this may also be satisfied, for example, by a third party organization making some level of commitment; I don't know what the exact language should be here, but it seems worth highlighting that 'just' two people is likely insufficient.

Reply:
I think it's implicitly true, but I also think there's value in explicitly upping the number, and in doing so, calling attention to higher expectations. Done.

I don't want to allow target maintainership to be diffused among an organization without pointing to specific people. The team members may rotate, but if the target maintainer team can't reliably maintain the required number of members, or if the members aren't responsive, that's a sign that the target isn't meeting the tier requirements.